eng/native/configurecompiler.cmake

```diff
@@ -417,25 +415,20 @@ endif(CLR_CMAKE_HOST_UNIX)
 if (MSVC)
   # Compile options for targeting windows
 
-  # The following options are set by the razzle build
   add_compile_options(/TP) # compile all files as C++
   add_compile_options(/nologo) # Suppress Startup Banner
   add_compile_options(/W3) # set warning level to 3
   add_compile_options(/WX) # treat warnings as errors
   add_compile_options(/Oi) # enable intrinsics
   add_compile_options(/Oy-) # disable suppressing of the creation of frame pointers on the call stack for quicker function calls
-  add_compile_options(/U_MT) # undefine the predefined _MT macro
-  add_compile_options(/GF) # enable read-only string pooling
   add_compile_options(/Gm-) # disable minimal rebuild
   add_compile_options(/Zp8) # pack structs on 8-byte boundary
   add_compile_options(/Gy) # separate functions for linker
-  add_compile_options(/Zc:wchar_t-) # C++ language conformance: wchar_t is NOT the native type, but a typedef
```

Comment:
It looks like `/Zc:wchar_t` is the default in the compiler. Are we good with moving to using the native `wchar_t` type instead of the typedef?

Reply:
I don't think we want the built-in type.

Reply:
Why not? 

I assume that the difference between built-in type and typedef'ed type only shows up in corner cases like duplicate type definition that will lead into compile errors. If it compiles, we should be good.

Reply:
The typical argument is linking against any Windows binaries since they compile with this turned off and it impacts the generated C++ symbol. This then leads to the argument/expectation that Microsoft ships binaries that have this turned off - which is always annoying since it isn't the default. If we change this now projects in the wild that are consuming assets can be broken since they may have been configured to disable the type projection. Think of the entire Visual Studio build environment.

I am not against it, but the decision does fly in the face of 20+ years of convention. Just want to make sure that is considered.

Reply:
Give the following from https://docs.microsoft.com/cpp/build/reference/zc-wchar-t-wchar-t-is-native-type

>We do not recommend /Zc:wchar_t- because the C++ standard requires that wchar_t be a built-in type. Using the typedef version can cause portability problems. If you upgrade from earlier versions of Visual Studio and encounter compiler error C2664 because the code is trying to implicitly convert a wchar_t to unsigned short, we recommend that you change the code to fix the error, instead of setting /Zc:wchar_t-.

I am inclined to retract my concern.